cms/auth/checks.py

```diff
@@ -0,0 +1,95 @@
+from collections.abc import Iterable
+from typing import TYPE_CHECKING, Any
+
+from django.conf import settings
+from django.core.checks import Error, register
+
+if TYPE_CHECKING:
+    from django.apps import AppConfig
+    from django.conf import LazySettings
+
+
+def _check_aws_cognito(errors: list, settings_obj: "LazySettings") -> None:
+    aws_cognito_enabled = getattr(settings_obj, "AWS_COGNITO_LOGIN_ENABLED", False)
+    if aws_cognito_enabled:
+        cognito_settings: list[tuple[str, str, str]] = [
+            ("AWS_COGNITO_USER_POOL_ID", "eu-west-2_xxxxxxxxx", "auth.E001"),
+            ("AWS_COGNITO_APP_CLIENT_ID", "the-cognito-app-client-id", "auth.E002"),
+            ("AWS_REGION", "eu-west-2", "auth.E003"),
+            ("AUTH_TOKEN_REFRESH_URL", "http://localhost:29500/tokens/self", "auth.E004"),
+            ("LOGOUT_REDIRECT_URL", "http://localhost:29500/logout", "auth.E005"),
+        ]
+        for setting, hint_value, error_id in cognito_settings:
+            value = getattr(settings_obj, setting, None)
+            if not value:
+                errors.append(
+                    Error(
+                        f"{setting} is required when AWS_COGNITO_LOGIN_ENABLED is True.",
+                        hint=f"Set {setting} to an appropriate value. e.g. '{hint_value}'.",
+                        id=error_id,
+                    )
+                )
```

Comment:
Let's return early:

```suggestion
def _check_aws_cognito(errors: list, settings_obj: "LazySettings") -> None:
    if not getattr(settings_obj, "AWS_COGNITO_LOGIN_ENABLED", False):
        return None
    cognito_settings: list[tuple[str, str, str]] = [
        ("AWS_COGNITO_USER_POOL_ID", "eu-west-2_xxxxxxxxx", "auth.E001"),
        ("AWS_COGNITO_APP_CLIENT_ID", "the-cognito-app-client-id", "auth.E002"),
        ("AWS_REGION", "eu-west-2", "auth.E003"),
        ("AUTH_TOKEN_REFRESH_URL", "http://localhost:29500/tokens/self", "auth.E004"),
        ("LOGOUT_REDIRECT_URL", "http://localhost:29500/logout", "auth.E005"),
    ]
    for setting, hint_value, error_id in cognito_settings:
        value = getattr(settings_obj, setting, None)
        if not value:
            errors.append(
                Error(
                    f"{setting} is required when AWS_COGNITO_LOGIN_ENABLED is True.",
                    hint=f"Set {setting} to an appropriate value. e.g. '{hint_value}'.",
                    id=error_id,
                )
            )
````